_posts/en/newsletters/2024-11-01-newsletter.md

```diff
@@ -0,0 +1,144 @@
+  of the parties involved in the timeout tree forfeit any of their funds
+  remaining in the tree---this incentivizes them to withdraw or find
+  alternative security for those funds before the expiry, which can
+  encourage the use of cheap offchain mechanisms rather than publishing
+  parts of the tree onchain.  In previously described timeout trees (see
+  [Newsletter #270][news270 timeout trees]), user funds that timed out
+  became the property of a service provider, but ZmnSCPxj reverses this
+  and makes a service provider's timed out funds become the property of
+  the users---this places the burden of getting transactions confirmed
+  on the service provider rather than end users.
+
+  The timeout trees used require each involved party to contribute a
+  signature.  This avoids the need for consensus changes but limits the
+  practical maximum number of users in a factory due to the well-known
+  [multi-signer coordination problem][news270 coordination].
+
+  Most of the leaves of the timeout tree are offchain funding
+  transactions for the common type of channel used today (LN-Penalty),
+  allowing some reuse of existing code for managing LN channels.  The
+  counterparties in each channel are an end user and the Lightning
+  service provider (LSP) who created the timeout tree.  Some of the
+  leaves of the tree may also be exclusively controlled by the LSP for
+  the purpose of rebalancing funds.
+
+  Between the root and the leaves are [duplex micropayment
+  channels][topic duplex micropayment channels].  Unlike LN-Penalty
+  channels, duplex channels allow more than two parties to safely share
+  funds; however, they also only allow a relatively small number of
+  state updates compared to LN-Penalty's effectively unlimited number of
+  updates.  The intermediate duplex channels are used to allow
+  rebalances involving the LSP and two end users; these rebalances could
+  securely complete at offchain speeds, allowing user Alice to receive
+  an incoming payment almost instantly even if she didn't previously
+  have enough capacity in her channel to accept it.
+
+  FIXME:illustration
+
+  In a [later post][zmnscpxj post2], ZmnSCPxj described replacing part
+  of a duplex channel with a [Spillman-style][spillman channel]
+  (simplex) micropayment channel.  This would be more efficient onchain
+  in the case of a cooperative close, although it would be less
+  efficient onchain in the case of a unilateral close.
+
+  The proposal received a moderate amount of discussion.  The author
+  said that one of the weaknesses of the proposal is its technical
+  complexity due to the use of multiple different channel types plus the
+  inherent challenge in any channel factory design of managing
+  additional offchain state.  However, it has the advantage of being
+  something that could be implemented by a single team and made
+  interoperable with the standard LN without requiring many changes to
```

Comment:
"Many" or "any"? If "many", what are the changes to the LN protocol that would be required?

Reply:
Many is correct based on my conversation with ZmnSCPxj in https://bitcoinops.org/en/podcast/2024/10/31/ ; quick rough manual transcription:

49:34 we do need some changes at the protocol layer.  What I'm planning
to propose is that we have some kind of plugable factory concept.  We're
gonig to use an open v1 message to open this channel and then we're
going to have a TLV that says, "hey this is the protocol that we're
using and this is an identifier that specifies the particular protocol
that we're using, and then afterwards you can say "at the factory
layer we did some messaging and we need to change the funding output of
this channel".